pkg/controller/node.go

```diff
@@ -84,11 +86,11 @@ func (r *ReconcileNode) Reconcile(ctx context.Context, request reconcile.Request
 	// If the node has no DeletionTimestamp, it means it's either create event or update event
 	if n.ObjectMeta.DeletionTimestamp.IsZero() && !n.Spec.Unschedulable {
 		// Add the new schedulable node in availableAttachmentsMap
-		r.addNodeToAvailableAttachmentsMap(ctx, n.Name, n.GetLabels())
-
-		// Node is schedulable, proceed to attempt creation of replica attachment
-		logger.Info("Node is now available. Will requeue failed replica creation requests.")
-		r.tryCreateFailedReplicas(ctx, nodeavailability)
+		if ok := r.addNodeToAvailableAttachmentsMap(ctx, n.Name, n.GetLabels()); ok {
```

Comment:
I think that even if the node is unscheduleable we should add it to the map to avoid skipping it in the future,

Reply:
Since we are reconciling both on updates to the schedulable property and updates to label changes, this will work but I think we are creating some indirect dependency here and this might cause an issue if we later change the update predicate.

Reply:
Do you mean `!n.Spec.Unschedulable` in line 87? Based on existing code, if the node becomes schedulable from unschedulable, it will be an Update event and add the node to map.

Reply:
if addNodeToAvailableAttachmentsMap() returns false here, should we do something like "return reconcile.Result{Requeue: true}, err" based on the err. So we can re-try addNodeToAvailableAttachmentsMap() if previously it failed because the label was not there yet.

Reply:
I'm worrying if the error is unresolvable, will this make an unlimited retry?

Reply:
Yes, I agree with you. No need to retry.

Reply:
I'll remove the `!n.Spec.Unschedulable` check before adding the node to the map.